### PR TITLE
feat Define specialOrthogonalGroup and specialUnitaryGroup

### PR DESCRIPTION
**Pull request by hmonroe**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/HalesJewett.lean [hunk 1]
```diff
@@ -214,8 +214,9 @@ theorem diagonal_apply {α ι} [Nonempty ι] (x : α) : Line.diagonal α ι x =
   simp_rw [Line.diagonal, Option.getD_none]
 #align combinatorics.line.diagonal_apply Combinatorics.Line.diagonal_apply
 
-/-- The Hales-Jewett theorem. This version has a restriction on universe levels which is necessary
-for the proof. See `exists_mono_in_high_dimension` for a fully universe-polymorphic version. -/
+/-- The **Hales-Jewett theorem**. This version has a restriction on universe levels which is
```
Comment (j-loreaux) [inline Mathlib/Combinatorics/HalesJewett.lean:217]: All these changes which only put theorem names in docstrings in bold face should be moved to a separate PR.

**Review by j-loreaux — COMMENTED** [review]
The docstring changes should be split into a separate PR, and we need to answer a question about how to proceed with the actual definition.

File: Mathlib/LinearAlgebra/UnitaryGroup.lean [hunk 2]
```diff
@@ -224,6 +224,18 @@ theorem mem_orthogonalGroup_iff' {A : Matrix n n β} :
...
+the inverse. and the determinant is one -/
+
+def specialOrthogonalGroup :=
+  { A : Matrix n n β // A ∈ Matrix.orthogonalGroup n β ∧ A.det = 1 }
+
+/-- `Matrix.specialUnitaryGroup n` is the group of `n` by `n` unitary matrices where determinant
+is one -/
+
+def specialUnitaryGroup :=
+  { A : Matrix n n β // A ∈ Matrix.unitaryGroup n β ∧ A.det = 1 }
```
[the remark this answers is not in the document]
Comment (hmonroe) [reply Mathlib/LinearAlgebra/UnitaryGroup.lean:237]: I will post a thread on Zulip
